pyvista/core/utilities/transform.py

```diff
@@ -51,93 +51,123 @@ class Transform(_vtk.vtkTransform):
 
     Examples
     --------
-    Concatenate two transformations with :meth:`translate` and :meth:`scale` using
-    post-multiplication (default).
+    Create two transformations using :meth:`translate` and :meth:`scale`.
 
+    >>> import numpy as np
     >>> import pyvista as pv
-    >>> transform = pv.Transform()
-    >>> transform.multiply_mode
-    'post'
     >>> position = (-0.6, -0.8, 2.1)
-    >>> scale = 2.0
-    >>> _ = transform.translate(position)
-    >>> _ = transform.scale(scale)
+    >>> translation = Transform().translate(position)
+    >>> translation.matrix
+    array([[ 1. ,  0. ,  0. , -0.6],
+           [ 0. ,  1. ,  0. , -0.8],
+           [ 0. ,  0. ,  1. ,  2.1],
+           [ 0. ,  0. ,  0. ,  1. ]])
+
+    >>> scale_factor = 2.0
+    >>> scaling = Transform().scale(scale_factor)
+    >>> scaling.matrix
+    array([[2., 0., 0., 0.],
+           [0., 2., 0., 0.],
+           [0., 0., 2., 0.],
+           [0., 0., 0., 1.]])
```

Comment:
Awesome job! Is it possible to assign `scale` to [`operator.__mul__`](https://docs.python.org/ja/3.10/library/operator.html#operator.__mul__)?

Reply:
ref [Python : Operator Overloading a specific type](https://stackoverflow.com/questions/3188666/python-operator-overloading-a-specific-type)

Reply:
ok 👍

We could probably overload `__add__` too to allow for translations if doing addition where `other` is array-like

Reply:
For things like `__mul__` and `__add__`, should we use those for post-multiply? e.g. `Transform() * 5`
But then use pre-multiply with `__rmul__`, `__radd__` ?  e.g. `5 * Transform()`?

Reply:
How about making it behave like NumPy's Array? I believe it could also use right multiplication as a scale. I do not fully understand the disadvantages of this, so if you have any additional information, please share :)

Reply:
The main thing to consider is that the transformations are not commutative, which is why there is pre- and post- multiply. The transformations use matrix operations by default, whereas numpy uses element-wise operations and broadcasting by default.

For matmul with `@`, the transform behaves exactly like a numpy array using pre-multiply. But there is no equivalent operator for post-multiply. So, with this PR as it is, `__add__` is used, but it is actually doing matrix multiplication for concatenating the transformations, not actual addition. This is very different from how a regular numpy array behaves. And overloading this to allow `__add__` where one array is 4x4 (the transform) and another array is 3x1 (position vector) also is very different from how a numpy array behaves, which would raise an error due to shape mis-match.

So I guess we should be cautious about making this class look too much like a numpy array to avoid unintended operations. For example, if we add a 4x4 numpy array to a Transform, should we expect regular numpy addition? Or should the 4x4 array be concatenated with the transform?

Reply:
> So I guess we should be cautious about making this class look too much like a numpy array to avoid unintended operations. For example, if we add a 4x4 numpy array to a Transform, should we expect regular numpy addition? Or should the 4x4 array be concatenated with the transform?

I can't provide a clear answer to this. I think it would be best to prohibit operations with objects other than Transform objects at the moment. If there is a request for operations with objects of other types at a later date, let's respond to it.

Reply:
> I think it would be best to prohibit operations with objects other than Transform objects at the moment.

To be clear, this PR _does_ allow operations with non-Transform objects, e.g. translation with a length-3 vector, scaling with a float or length-3 vector, or concatenation with a 3x3 or 4x4 NumPy array or vtkMatrix (i.e. any TransformLike object). E.g. see the parameters for this test:
https://github.com/pyvista/pyvista/blob/c9429d272a894c899efc24f1e0d697c29518e0df/tests/core/test_utilities.py#L1419-L1422

But, unlike with #6508, _only_ the `Transform` overloads are used for the operation, not `np.array` overloads. So, if a `Transform` object is either on the right or left of the operator, then a `Transform` operation is always performed (not a numpy array operation)